katprep/snapshot.py

```diff
@@ -57,240 +51,272 @@ def parse_options(args=None):
-                    pass
-
-            #get owner
-            try:
-                SYSTEM_ERRATA[system["name"]]["params"]["owner"] =  \
-                    SAT_CLIENT.get_name_by_id(params_obj["owner_id"], "user")
-            except SessionException:
-                #no user
-                SYSTEM_ERRATA[system["name"]]["params"]["owner"] = ""
-                LOGGER.debug("No owner for system '%s' defined", system["name"])
-
-            #set HW flag
-            if not params_obj["facts"]["is_virtual"].lower() == "true":
-                SYSTEM_ERRATA[system["name"]]["params"]["system_physical"] = True
-
-            #add errata information if applicable
-            if int(errata_counter["total"]) > 0:
-                result_obj = json.loads(
-                    SAT_CLIENT.api_get("/hosts/{}/errata".format(system["id"]))
-                )
-                SYSTEM_ERRATA[system["name"]]["errata"] = result_obj["results"]
-                #remove _all_ the reboot suggested flags as Pandoc is too dump
-                #to check the value
-                for errata in SYSTEM_ERRATA[system["name"]]["errata"]:
-                    if not errata["reboot_suggested"]:
-                        del errata["reboot_suggested"]
-        except KeyError as err:
-            LOGGER.error(
-                "Unable to get system information for '%s', " \
-                "dropping system!", system["name"])
-            pass
-        except ValueError as err:
-            LOGGER.info("Unable to get data: '%s'", err)
-
-
-
-def create_report():
-    """Creates a JSON report including errata information of all hosts."""
-
-    try:
-        with open(OUTPUT_FILE, 'w') as target:
-            target.write(json.dumps(SYSTEM_ERRATA))
-    except IOError as err:
-        LOGGER.error("Unable to store report: '%s'", err)
+            with open(OUTPUT_FILE, "w") as target:
+                target.write(json.dumps(system_info))
+        except IOError as err:
+            LOGGER.error("Unable to store report: '%s'", err)
+        else:
+            LOGGER.info("Report '%s' created.", OUTPUT_FILE)
     else:
-        LOGGER.info("Report '%s' created.", OUTPUT_FILE)
-
+        LOGGER.info("Empty report - please check.")
 
 
 def main(options, args):
-    """Main function, starts the logic based on parameters."""
-    global SAT_CLIENT, OUTPUT_FILE
+    """
+    Main function, starts the logic based on parameters
+    """
+    global MGMT_CLIENT, OUTPUT_FILE
 
     LOGGER.debug("Options: %s", options)
     LOGGER.debug("Arguments: %s", args)
 
-    #set output file
+    # set output file
     if options.output_path == "":
         options.output_path = "./"
-    elif options.output_path != "" and \
-    options.output_path[len(options.output_path)-1:] != "/":
-        #add trailing slash
+    elif (
+            options.output_path != ""
+            and options.output_path[len(options.output_path) - 1:] != "/"
+    ):
```

Comment:
This can be done a lot simpler:

```suggestion
    elif not options.output_path.endswith("/"):
```